File: [BOJ] 백준 풀이/단계별로 풀어보기 (step by step)/#8 기본 수학 1/10250_ACM 호텔 (22.02.03 성공).cpp
#include <bits/stdc++.h>

using namespace std;

int main() {
    ios::sync_with_stdio(0);
    cin.tie(0);

    int T, H, W, N;

    cin >> T;

    for (int i = 0; i < T; ++i) {
        cin >> H >> W >> N; // 높이, 길이, 방 출력

        int floor = N % H; // 머무를 층
        int room = N / H + 1; // 머무를 방

        if (floor == 0) floor = H;
        if (N % H == 0) room -= 1;

        cout << floor * 100 + room << "\n";
    }
}

File: 백준 풀이/~23.12.31/#07 기본 수학 1/10250_ACM 호텔 (22.02.03 성공).py
T = int(input()) # 테스트 케이스

while T:
    H, W, N = map(int, input().split())
    floor = N % H
    room = N // H + 1

    if floor == 0:
         floor = H
    if N % H == 0:
         room -= 1

    print(f"{floor * 100 + room}")

    T -= 1